plan/physical_plan_builder.go

```diff
@@ -619,6 +627,11 @@ func (p *Join) convert2IndexNestedLoopJoinLeft(prop *requiredProperty, innerJoin
 	if lInfo.p == nil {
 		return nil, nil
 	}
+	// If the outer table's row count is reliable and don't exceed the MaxRowCountForINLJ or we use hint to force
+	// choosing index nested loop join, we will continue building. Otherwise we just break and return nil.
+	if !forced && (!lInfo.reliable || lInfo.count > uint64(p.ctx.GetSessionVars().MaxRowCountForINLJ)) {
```

Comment:
The logic to check whether to build nested loop join is scattered in multiple places, can we extract one method to do that?